packages/react-router-dev/vite/plugin.ts

```diff
@@ -616,17 +606,19 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = () => {
     ${Object.keys(routes)
       .map((key, index) => {
         let route = routes[key]!;
-        if (routeIdsToImport.has(key)) {
+        if (isSpaMode && key !== "root") {
```

Comment:
We can simplify this a bit - we don't even need to include the index route or any layouts below root - we only render the root route and it's `HydrateFallback` so that any application paths can hydrate using the same file (see https://github.com/remix-run/react-router/blob/fbea1eaa754522f8be0bcc5073df6bc2d9c34948/packages/react-router/lib/dom/ssr/components.tsx#L193)